### DISCUSSION TITLE
Country Not Mentioned in 2FA SMS authentication

### DISCUSSION THREAD

**Discussion by CodeMode365** [community/community#65154] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi, I'm from Nepal and recently a 2FA notification popped up on my GitHub profile and I tried to do so. Then when I tried to create 2FA through SMS my country was not mentioned. Please consider adding my country code. 
My country is **Nepal(+977)**

<img width="928" alt="image" src="https://github.com/community/community/assets/102400760/7dad75e9-3c47-41db-b62b-96912dd0070e">

**Comment by nirav-gajera** [comment]
maybe nepal's code is not listed in github 

In the meantime, you can still set up 2FA using an authenticator app like Google Authenticator or Authy. These apps generate a time-based one-time password that you can use to authenticate yourself to GitHub.

for more you can contact [Github Support](https://support.github.com/)

**Comment by mikenjuki** [comment]
I also had the same issue. The thing is, **GitHub doesn't support sending SMS messages to every country**.
So until then, your best option is to use the other 2FA methods provided. 
I hope this helps.

**Comment by ghostinhershell** [comment]
Hi @CodeMode365 👋 

Thanks so much for reaching out. If your country is not on [the list](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported#supported-countries-for-sms-authentication), then we aren't currently able to reliably deliver text messages to your country. This can change in the future and will update this list periodically.

If GitHub doesn’t support two-factor authentication via text message for your country of residence, you can set up authentication via a TOTP application like the ones @nirav-gajera mentioned above. For more information on how to configure 2FA, please review our [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).

**Comment by jrosadob** [comment]
Mi país tampoco está en la lista, Perú (+51)
Deberían tener otros medios adicionales como el correo por ejemplo.
En mi pais los celulares son muy faciles de "perder" lo ideal sería usar el correo, o SMS es relativamente facil recuperar el numero de telefono, pero recuperar la app de authenticacion es complicado.

**Comment by lucioduran** [comment]
neither Uruguay +598

**Comment by metux** [comment]
Little script to run totp entirely from your local machine (w/o extra devices)

https://gist.github.com/metux/ce43da58d9759019a0fbf4d25ccb1b00

**Comment by welydev** [comment]
It seems no french territories except France are supported either. (Martinique +596, Guadeloupe +590, Guyane +594, La Réunion +262, etc...)

**Comment by dev1joe** [comment]
neither Egypt +20

**Comment by araf-15** [comment]
Please Add 
Bangladesh (+880)
 also

**Reply by bijoy26** [reply]
Second this! Most of the other major providers are supporting SMS gateway to Bangladesh for 2FA auth.

**Comment by FabuTheFool** [comment]
neither Paraguay +595

**Comment by the-ripper77** [comment]
GitHub is still not supporting 2FA SMS for +977

**Comment by manavpatil-dev0** [comment]
**GitHub's SMS support for 2FA can vary by country and region, so this may be related to SMS delivery availability rather than Nepal simply being missing from the UI.

In the meantime, I'd strongly recommend using a TOTP authenticator app or a passkey instead of SMS for 2FA. Still, adding support for Nepal (+977) would definitely improve accessibility for GitHub users in the region.**

**Comment by dpaguba** [comment]
The reason your country is not selectable is that GitHub can only offer SMS 2FA where it can reliably deliver text messages, and that list is limited. You can see the current supported countries here: https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported. If your code is not on it, SMS is not available to you yet. The list does get updated over time, so it is worth rechecking.

If you specifically need SMS, a workaround some people use is an eSIM with a number from a supported country. I would treat that as a last resort though: it costs money to keep active, and if the number ever lapses you can lock yourself out of the account. An authenticator app or passkey avoids all of that.
